terracotta/server/fields.py

```
@@ -45,3 +47,16 @@ def validate_stretch_range(data: Any) -> None:
     if isinstance(data, str):
         if not re.match("^p\\d+$", data):
             raise ValidationError("Percentile format is `p<digits>`")
+
+
+def validate_color_transform(data: Any) -> None:
+    """
+    Validate that the color transform is a string and can be parsed by `color_operations`.
+    """
+    if not isinstance(data, str):
+        raise ValidationError("Color transform needs to be a string")
+
+    try:
+        parse_operations(data)
```

If that's the case, should we try and use the color transform here on a trivial array (such as `np.ones((1, 1))`)?

---

We could do that, it would get slightly ugly because the success of applying the transform will depend on whether it's applied on a singleband or rgb image. And `validate_color_transform` is only called from marshmallow.

I can use partial to suggest to `validate_color_transform` what the array it should test against.

---

Ok maybe I misunderstood. What is being validated here? In which case can a transformation string pass this validation and fail later during application?

---

So to answer your question more fully, it does provide some validation but the parser has some shortcomings. Apart from the example I gave it will also let you pass negative numbers.

---

Ok, so that's why I suggested we actually try and apply the operation to some fake data here (without checking the result). If someone passed an invalid parameter it will raise an exception, and if not the code will continue.

---

Yes, I was mainly talking about the implementation of that idea. 

I think it works nicely now, I pushed another update.